### PR TITLE
Feat/progressbar valueformatter

### PR DESCRIPTION
**Pull request by jsheely**
This is related to pr https://github.com/spectreconsole/spectre.console/pull/1388 and issue https://github.com/spectreconsole/spectre.console/issues/1382

Goal is to provide formatting for number values in the progress bar

- [x] I have read the [Contribution Guidelines](../CONTRIBUTING.md)
- [x] I have commented on the issue above and discussed the intended changes
- [x] A maintainer has signed off on the changes and the issue was assigned to me
- [x] All newly added code is adequately covered by tests
- [x] All existing tests are still running without errors
- [x] The documentation was modified to reflect the changes _OR_ no documentation changes are required.

### CODE REVIEW

File: test/Spectre.Console.Tests/Unit/Widgets/ProgressBarTests.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            Width = 60,
+            Value = 9000,
+            MaxValue = 9000,
+            ShowValue = true,
+            ValueFormatter = (value, _) => value.ToString("N0"),
+        };
+
+        // When
+        console.Write(bar);
+        // Then
```
Comment (patriksvensson) [inline test/Spectre.Console.Tests/Unit/Widgets/ProgressBarTests.cs:40]: Space between when and then clause.

File: test/Spectre.Console.Tests/Unit/Widgets/ProgressBarTests.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    public async Task Should_Render_Correctly()
+    {
+        // Given
+        var console = new TestConsole();
+
+        var bar = new ProgressBar() { Width = 60, Value = 9000, MaxValue = 9000, ShowValue = true };
+
+        // When
+        console.Write(bar);
+        // Then
```
Comment (patriksvensson) [inline test/Spectre.Console.Tests/Unit/Widgets/ProgressBarTests.cs:18]: Space between when and then clause.

File: test/Spectre.Console.Tests/Unit/Widgets/ProgressBarTests.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+public class ProgressBarTests
+{
+    [Fact]
+    [Expectation("Render")]
+    public async Task Should_Render_Correctly()
+    {
+        // Given
+        var console = new TestConsole();
+
+        var bar = new ProgressBar() { Width = 60, Value = 9000, MaxValue = 9000, ShowValue = true };
```
Comment (patriksvensson) [inline test/Spectre.Console.Tests/Unit/Widgets/ProgressBarTests.cs:14]: Put each property on each own line.

**Review by patriksvensson — CHANGES_REQUESTED** [review]
Looks good to me, but we should probably add an extension as well so it's consistent with other widgets yhat supports custom formatting.

We will also need to update the documentation somit reglects the new additions.

File: src/Spectre.Console/Extensions/BarChartExtensions.cs [hunk 4]
```diff
@@ -116,6 +116,37 @@ public static BarChart AddItems<T>(this BarChart chart, IEnumerable<T> items, Fu
...
+        chart.ValueFormatter = func;
+        return chart;
+    }
+
+    /// <summary>
+    /// Tags will be shown.
+    /// </summary>
+    /// <param name="chart">The breakdown chart.</param>
+    /// <param name="func">The value formatter to use.</param>
+    /// <returns>The same instance so that multiple calls can be chained.</returns>
```
Comment (patriksvensson) [inline src/Spectre.Console/Extensions/BarChartExtensions.cs:135]: Documentation does not reflect the method.

File: src/Spectre.Console/Extensions/BarChartExtensions.cs [hunk 5]
```diff
@@ -116,6 +116,37 @@ public static BarChart AddItems<T>(this BarChart chart, IEnumerable<T> items, Fu
         return chart;
     }
 
+    public static BarChart UseValueFormatter(this BarChart chart, Func<double, CultureInfo, string>? func)
```
Comment (patriksvensson) [inline src/Spectre.Console/Extensions/BarChartExtensions.cs:119]: No documentation

**Review by patriksvensson — CHANGES_REQUESTED** [review]
Looks good but docs need to be updated.

**Comment by jsheely** [comment]
Not sure I know what this Pull Request / Documentation error is about

```
Error: Inputs/Process » ExecuteIf » ExtractFrontMatter » ParseYaml » [/home/runner/work/spectre.console/spectre.console/docs/input/widgets/breakdownchart.md => widgets/breakdownchart.md] (Line: 4, Col: 13, Idx: 126) - (Line: 4, Col: 13, Idx: 126): Block sequence entries are not allowed in this context.
```

**Review by patriksvensson — APPROVED** [review]

**Comment by patriksvensson** [comment]
Merged! Thank you for your contribution. Much appreciated! 👍

**Pull request merged**